Backpack-SwiftUI/Calendar/Classes/Range/RangeCalendarContainer.swift

```diff
@@ -0,0 +1,237 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+protocol RangeCalendarSelectionHandler {
```

Comment:
Introducing this protocol so that we can extract the selection handling logic from the Container view and eventually swap it with a new implementation, which is a change that's coming for the range selection calendars, which will be experimented upon.

When the new seleciton logic is implemented, we should only add a new class that conforms to this protocol and, if flag is enabled use one or the other implementaion.